models/core-java/src/main/java/org/commonjava/indy/stats/IndyDeprecatedApis.java

```diff
@@ -88,7 +88,16 @@ public DeprecatedApiEntry getDeprecated( String reqApiVersion )
         {
             return null;
         }
-        Float reqVer = Float.parseFloat( reqApiVersion );
+        Float reqVer;
+        try
+        {
+            reqVer = Float.parseFloat( reqApiVersion );
+        }
+        catch ( NumberFormatException e )
```

Comment:
@ligangty can be also null. So maybe checking for NullPointerException also is god thig?

Reply:
@geored I think this null is checked in caller code, see ApiVersionFilter.doFilter. 
But as you said, I agree that return null is really a code smell. I think I should refactor this by using "Optional"  way.